sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1291,4 +1291,14 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       Seq(1 -> "a").toDF("i", "j").filter($"i".cast(StringType) === "1"),
       Row(1, "a"))
   }
+  
+  test("SPARK-12982: Add table name validation in temp table registration") {
```

Comment:
@hvanhovell @davies 
Shall I include the changes for the above review comments now ?